### DISCUSSION TITLE
Eslint prettier rules not firing in .md files

### DISCUSSION THREAD

**Discussion by jdimmerman** [evidence-dev/evidence#2383] [General]
## Steps To Reproduce

With my eslint configuration, which includes prettier, the eslint rules themselves are firing on the javascript but the prettier rules are not.

Some notes
* This also is likely unrelated to the VSCode extension but I'm unclear where to post this issue.
* This may actually be a feature request rather than an issue. 
* I've tried by executing the eslint cli (`npx eslint .`) and via the VSCode eslint extension
* I had to include the `mdx` eslint plugin so that frontmatter didn't cause linting failures, but even without this plugin I was unable to get prettier rules to fire

I've posted my eslint config below and then these screenshots to illustrate the issue. Notice
* In the .svelte file, both the `eslint/no-undef` and `prettier/prettier` rules are firing
* In the .md file, only the `eslint/no-undef` rule is firing

![image](https://github.com/evidence-dev/evidence-vscode/assets/12055704/3c552283-745d-470a-889d-c8ada0ebea15)
![image](https://github.com/evidence-dev/evidence-vscode/assets/12055704/8f51c081-c5cb-47c6-8cd6-c3587559c91f)

### Environment

- VSCode Extension version (Check in VSCode Marketplace): v1.1.0
- Node version (`node -v`): 19.7
- npm version (`npm -v`): 9.5
- Database: snowflake

.eslintrc.cjs
```cjs
module.exports = {
  env: {
    browser: true,
    es2021: true,
  },
  extends: ['standard', 'prettier', 'plugin:svelte/recommended'],
  parserOptions: {
    ecmaVersion: 'latest',
    sourceType: 'module',
  },
  plugins: ['prettier', 'mdx'],
  rules: {
    'prettier/prettier': ['error'],
  },
  overrides: [
    {
      files: '*.svelte',
      rules: {
        'no-unused-expressions': 'off',
        'no-sequences': 'off',
        'no-use-before-define': 'off',
        camelcase: 'off',
      },
    },
    {
      files: '*.md',
      parser: 'svelte-eslint-parser',
      // processor: 'svelte/.svelte', // this did not help
      extends: ['plugin:mdx/recommended'], // commenting this out causes a syntax error around frontmatter but did not cause the prettier rules to fire
      rules: {
        'no-unused-expressions': 'off',
        'no-sequences': 'off',
        'no-use-before-define': 'off',
        camelcase: 'off',
      },
    },
  ],
  globals: {
    Column: true,
    Tab: true,
    Tabs: true,
  },
};
```

.prettierrc
```json
{
    "semi": true,
    "singleQuote": true,
    "tabWidth": 2
}
```

## Expected Behavior

Eslint and prettier rules fire in .md files, similar to .svelte files

## Actual Behaviour

Only Eslint rules fire in the .md files and prettier rules do not.

## Workarounds

Don't use prettier in .md files.

**Comment by RandomFractals** [comment]
@jdimmerman thank you for posting these details.

we'll try to reproduce your current linters config in a week or so, as we are taking some time off after this brisk tooling run to get some cool new basic commands for you to use on Evidence dev projects in vscode.

Your notes are stellar and we might have a solution for that markdown related linting problem later on.

Also, good to see that you are testing it with `mdx` too. It's one of the well designed and supported markdown extensions ecosystems out there.

**Comment by jdimmerman** [comment]
@RandomFractals that sounds great and I get that... I hope that you have a good and relaxing week and thanks!

**Comment by jdimmerman** [comment]
Hey @RandomFractals - Just following up here, thanks!

**Comment by archiewood** [comment]
~~@hughess~~

**Comment by archiewood** [comment]
And @ItsMeBrianD  - I'm guessing this should be in the main evidence repo? WDYT?

**Comment by ItsMeBrianD** [comment]
Yeah I think the main repo is the best place to address this
